src/test/unit/test_spans_container.py

```diff
@@ -204,6 +204,51 @@ def test_add_tag():
     ]
 
 
+def test_start_manual_trace_simple_flow():
+    SpansContainer.get_span().start_manual_trace("11")
+    SpansContainer.get_span().stop_manual_trace("11")
+
+    manual_tracers = SpansContainer.get_span().function_span[MANUAL_TRACES_KEY]
+    now = datetime.now().timestamp() * 1000
+    one_min_ago = (datetime.now() - timedelta(minutes=1)).timestamp() * 1000
+    assert len(manual_tracers) == 1
+    assert manual_tracers[0]["startTime"] >= one_min_ago
+    assert manual_tracers[0]["startTime"] <= now
+    assert manual_tracers[0]["endTime"] >= one_min_ago
+    assert manual_tracers[0]["endTime"] <= now
+
+
+def test_end_manual_trace_name_not_exist():
+    SpansContainer.get_span().stop_manual_trace("11")
+
+    manual_tracers = SpansContainer.get_span().function_span[MANUAL_TRACES_KEY]
+    assert manual_tracers == []
+
+
+def test_start_manual_trace_start_twice():
+    SpansContainer.get_span().start_manual_trace("11")
+    SpansContainer.get_span().start_manual_trace("11")
+    SpansContainer.get_span().stop_manual_trace("11")
+
+    manual_tracers = SpansContainer.get_span().function_span[MANUAL_TRACES_KEY]
+    assert len(manual_tracers) == 1
+
+
+def test_start_manual_trace_multiple():
+    SpansContainer.get_span().start_manual_trace("11")
+    SpansContainer.get_span().start_manual_trace("22")
+    SpansContainer.get_span().stop_manual_trace("11")
+    SpansContainer.get_span().stop_manual_trace("22")
+
+    manual_tracers = SpansContainer.get_span().function_span[MANUAL_TRACES_KEY]
+    now = datetime.now().timestamp() * 1000
+    one_min_ago = (datetime.now() - timedelta(minutes=1)).timestamp() * 1000
+    assert len(manual_tracers) == 2
+    for manual_trace in manual_tracers:
+        assert manual_trace["startTime"] >= one_min_ago
+        assert manual_trace["startTime"] <= now
```

Comment:
here again you can simplify and make the assertion much stronger IMO (e.g. check that the startTime of the 22 is lower than endTime of 11